src/protocol/xproto.rs

```diff
@@ -6225,23 +6225,35 @@ impl<'input> CreateWindowRequest<'input> {
         Ok((vec![request0.into(), value_list_bytes.into(), padding0.into()], vec![]))
     }
     /// Parse this request given its header, its body, and any fds that go along with it
-    pub fn try_parse_request(header: RequestHeader, body: &'input [u8]) -> Result<Self, ParseError> {
-        validate_request_pieces(header, body, Some(CREATE_WINDOW_REQUEST), None)?;
-        // TODO: deserialize depth
-        // TODO: deserialize wid
-        // TODO: deserialize parent
-        // TODO: deserialize x
-        // TODO: deserialize y
-        // TODO: deserialize width
-        // TODO: deserialize height
-        // TODO: deserialize border_width
-        // TODO: deserialize class
-        // TODO: deserialize visual
-        // TODO: deserialize value_mask
-        // TODO: deserialize value_list
-        let _ = body;
-        // TODO: produce final struct
-        unimplemented!()
+    pub fn try_parse_request(header: RequestHeader, value: &'input [u8]) -> Result<Self, ParseError> {
+        validate_request_pieces(header, value, Some(CREATE_WINDOW_REQUEST), None)?;
+        let (depth, remaining) = u8::try_parse(&[header.minor_opcode])?;
```

Comment:
Actually... can this be simplified? This is a quite complicated way to turn an `u8` into an `u8`. Since you already assert that all the "inside the header" fields have length 1... are we perhaps lucky and they are all of type `CARD8`? If so, this code could be simplified.

Reply:
I'm sure some of them are `BOOL`. I haven't checked but while it's ugly I suspect LLVM will chew it up and do the right thing in an optimized build.

Reply:
Okay